lib/public/services/detectors/dplDetectorsProvider.js

```diff
@@ -30,7 +29,15 @@ export class DplDetectorsProvider extends Observable {
         this._physical$ = ObservableData.builder()
             .source(this._all$)
             .apply((remoteDetectors) => remoteDetectors.apply({
-                Success: (detectors) => detectors.filter(({ name }) => name && name !== TST_DETECTOR_NAME),
+                Success: (detectors) => {
+                    const filteredDetectors = detectors.filter(({ name }) => name && name.toUpperCase() !== VIRTUAL_DETECTOR_NAME.TST);
+                    const gloIndex = filteredDetectors.findIndex(({ name }) => name.toUpperCase() === VIRTUAL_DETECTOR_NAME.GLO);
+                    if (gloIndex !== -1) {
+                        const [gloDetector] = filteredDetectors.splice(gloIndex, 1);
+                        return [gloDetector, ...filteredDetectors.sort(({ name: nameA }, { name: nameB }) => nameA < nameB)];
+                    }
+                    return filteredDetectors.sort();
```

Comment:
You are currently iterating 5 times over these arrays with the `filter`, `findIndex`, `splice`, 2x`sort`
I would recommend you improve it with 2 iterations:
- 1 `for` in which you build a new array which does not contain TST or GLO. In the same for you identify the index of `GLO`
- 1 sort of the newly created array + return of GLO if present

Reply:
refactored